tests/Shared/Playwright/PlaywrightProvider.cs

```diff
@@ -0,0 +1,76 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.InteropServices;
+using Microsoft.Playwright;
+
+namespace Aspire.Workload.Tests;
+
+public class PlaywrightProvider
+{
+    public const string BrowserPathEnvironmentVariableName = "BROWSER_PATH";
+    private const string PlaywrightBrowsersPathEnvironmentVariableName = "PLAYWRIGHT_BROWSERS_PATH";
+
+    public static async Task<IBrowser> CreateBrowserAsync(BrowserTypeLaunchOptions? options = null)
+    {
+        var playwright = await Playwright.CreateAsync();
+        string? browserPath = Environment.GetEnvironmentVariable(BrowserPathEnvironmentVariableName);
+        if (!string.IsNullOrEmpty(browserPath) && !File.Exists(browserPath))
+        {
+            throw new FileNotFoundException($"Browser path {BrowserPathEnvironmentVariableName}='{browserPath}' does not exist");
+        }
+
+        options ??= new() { Headless = true };
+        options.ExecutablePath ??= browserPath;
+
+        if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX) && string.IsNullOrEmpty(browserPath))
+        {
+            var probePath = "/Applications/Google Chrome.app/Contents/MacOS/Google Chrome";
+            if (File.Exists(probePath))
+            {
+                options.ExecutablePath = probePath;
+            }
+        }
+        return await playwright.Chromium.LaunchAsync(options).ConfigureAwait(false);
+    }
+
+    // Tries to set PLAYWRIGHT_BROWSERS_PATH to the location of the playwright-deps directory in the repo
+    public static void DetectAndSetInstalledPlaywrightDependenciesPath(DirectoryInfo? repoRoot = null)
+    {
+        if (!string.IsNullOrEmpty(Environment.GetEnvironmentVariable(PlaywrightBrowsersPathEnvironmentVariableName)))
+        {
+            // this would be the case for helix where the path is set to a
+            // payload directory
+            return;
+        }
+
+        if (repoRoot is null)
```

Comment:
Does this case ever happen? If repoRoot is null its because we are in Helix, right?

Reply:
This one is if the caller didn't pass in a pre-computed `repoRoot` then we do the detection here. In `BuildEnvironment.cs` we already have `repoRoot` computed so this would avoid it being repeated. But when using it outside the workload tests, the caller might not have it, and thus pass `null` here.

But now that you mention it, I'm repeating this code in BuildEnvironment.cs . This should become a separate method .. somewhere!

Reply:
Extract that to a `TestUtils.FindRepoRoot`.